### PR TITLE
Add signing key manager to Vault

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes https://github.com/google/exposure-notifications-server/issues/915

⚠️ Vault doesn't support destroying a single key version, so it returns an error.

**Release Note**

```release-note
Add support for SigningKeyManager interface with Vault
```

### CODE REVIEW

File: pkg/keys/hashicorp_vault.go [hunk 1]
```diff
@@ -72,6 +75,144 @@ func NewHCValueKeyID(keyID string) (*HCValueKeyID, error) {
...
 }
 
+type readKeyResponse struct {
+	Versions map[string]struct {
+		CreationTime time.Time `ms:"creation_time"`
+		PublicKeyPEM string    `ms:"public_key"`
+	} `ms:"keys"`
+}
+
+type vaultKeyVersion struct {
```
Comment (mikehelmick) [inline pkg/keys/hashicorp_vault.go:85]: compile time cast assert?

Comment (mikehelmick) [reply pkg/keys/hashicorp_vault.go:85]: i know it’s not necessary - but we normally do it.. so up to you

Comment (sethvargo) [reply pkg/keys/hashicorp_vault.go:87]: I didn't originally add it because the return type from the method enforces it at compile time. Added.

Comment (mikehelmick) [reply pkg/keys/hashicorp_vault.go:87]: yeah - just trying to be internally consistent with what we've done elsewhere. Thanks!

**Pull request merged**